### ISSUE TITLE
15 second delay after code using node-phantom exits

### ISSUE THREAD

**Issue by ewaters** [alexscheelmeyer/node-phantom#17]
I've written a quick script to illustrate this problem.  I'm not sure what's happening, as I've been able to confirm that phantomjs exits when phantom.exit() callback hits (see below).  All the tests in the distribution, run via expresso, also suffer from this 15 second delay.

I've tried node v0.6.21, v0.7.12 and v0.8.14 (what I use).  I've tried phantomjs v0.6.1 and v0.7.0 (what I use).  This is on a CentOS 6 x86_64 box.

```
date && node slow_test.js && date && node -v && phantomjs --version
Fri Nov 16 15:45:20 EST 2012
16 Nov 15:45:22 - Phantom created
16 Nov 15:45:22 - ewaters  28965 20.0  0.1 1415240 54104 pts/32  Sl+  15:45   0:00 phantomjs /home/ewaters/code/Absinthe/node_modules/node-phantom/bridge.js 43012
ewaters  28973  0.0  0.0 106088  1156 pts/32   S+   15:45   0:00 /bin/sh -c ps aux | grep phantomjs
ewaters  28975  0.0  0.0 103240   872 pts/32   S+   15:45   0:00 grep phantomjs

16 Nov 15:45:22 - Page created
16 Nov 15:45:22 - Page opened with status success
16 Nov 15:45:22 - Phantom exited
16 Nov 15:45:22 - ewaters  28979  0.0  0.0 106088  1156 pts/32   S+   15:45   0:00 /bin/sh -c ps aux | grep phantomjs
ewaters  28981  0.0  0.0 103240   872 pts/32   S+   15:45   0:00 grep phantomjs

16 Nov 15:45:22 - Waterfall complete
Fri Nov 16 15:45:37 EST 2012
v0.8.14
1.7.0
```

``` javascript
var phantom = require('node-phantom'),
    async   = require('async'),
    util    = require('util'),
    child_process = require('child_process');

var page, ph;

async.waterfall([
  function (cb) {
    phantom.create(cb);
  },
  function (result, cb) {
    ph = result;
    util.log("Phantom created");
    child_process.exec("ps aux | grep phantomjs", cb);
  },
  function (stdout, stderr, cb) {
    util.log(stdout);
    ph.createPage(cb);
  },
  function (result, cb) {
    page = result;
    util.log("Page created");
    page.open("http://www.google.com", cb);
  },
  function (status, cb) {
    util.log("Page opened with status " + status);
    ph.exit(cb);
  },
  function (cb) {
    util.log("Phantom exited");
    child_process.exec("ps aux | grep phantomjs", cb);
  },
  function (stdout, stderr, cb) {
    util.log(stdout);
    cb();
  }
], function (err) {
  if (err) util.log(err);
  util.log("Waterfall complete");
});
```

**Comment by alexscheelmeyer** [comment]
I agree that it seems sluggish to shut down after calling exit. Have you tried the pull request with the phantom.kill invocation? It is not working for me, but could perhaps be the thing needed on linux systems (I am on OS X).

**Comment by lobsang** [comment]
I am having the same 15 seconds problem on OS X. I am using node 0.8.17 and phantomjs 1.8.

**Comment by alexscheelmeyer** [comment]
This seems to now be fixed, through pull request #28 that I just merged.
